index.js

```diff
@@ -166,22 +165,20 @@ class HandlebarsRenderer {
      * @param {Object} A set of templates to register with handlebars
      */
     addTemplates(templates) {
-        _.each(templates, (template, path) => {
-            // Don't do this work twice, first one wins.
-            if (typeof this.handlebars.partials[path] !== 'undefined') {
+        const keys = Object.keys(templates);
+
+        for (let i = 0; i < keys.length; i++) {
+            const template = keys[i];
+            if (typeof this.handlebars.partials[template] !== 'undefined') {
                 return;
             }
-
-            // Check if it is a precompiled template
             try {
-                template = this._tryRestoringPrecompiled(template);
+                templates[template] = this._tryRestoringPrecompiled(templates[template]);
```

Comment:
Let's not mutate the incoming parameter. `template` should be a local variable declared inside the loop.

Reply:
fixed.